### DISCUSSION TITLE
"non-conforming drawing primitive definition" error, even when using official example. Inkscape renderer?

### DISCUSSION THREAD

**Discussion by matanshukry** [ImageMagick/ImageMagick#2583] [Help] [answered]
I'm trying to draw a primitive as an extra image and I got an error.
To make sure it's not an error in my command, I went to the documentation and found this example ("converted" to windows):

```
magick -size 320x90 canvas:none -stroke snow4 -size 1x90 -tile gradient:white-snow4  -draw 'roundrectangle 16, 5, 304, 85 20,40' +tile -fill snow  -draw 'roundrectangle 264, 5, 304, 85  20,40' -tile gradient:chartreuse-green  -draw 'roundrectangle 16,  5, 180, 85  20,40' -tile gradient:chartreuse1-chartreuse3 -draw 'roundrectangle 140, 5, 180, 85  20,40' +tile -fill none -draw 'roundrectangle 264, 5, 304, 85 20,40' -strokewidth 2 -draw 'roundrectangle 16, 5, 304, 85 20,40' ( +clone -background snow4 -shadow 80x3+3+3 ) +swap -background none -layers merge ( +size -pointsize 90 -strokewidth 1 -fill red label:'50 %' -trim +repage ( +clone -background firebrick3 -shadow 80x3+3+3 ) +swap -background none -layers merge ) -insert 0 -gravity center -append -background white -gravity center -extent 320x200 cylinder_shaded.png
```

However I'm getting this error for it:
``magick: non-conforming drawing primitive definition `roundrectangle' @ error/draw.c/RenderMVGContent/4479.``

I found another discussion talking about MVG renderer vs Inkscape, but I have Inkscape installed (1.0) and I can't find any option to enable the Inkscape renderer?

p.s.
I also tried adding -verbose to the command but it doesn't add any output

**Comment by GeeMack** [comment, marked as answer]
It looks like you've assembled the draw operations incorrectly. You have this...

    ... -draw 'roundrectangle 16, 5, 304, 85 20,40' ...

You should have something more like this...

    ... -draw 'roundrectangle 16,5 304,85 20,40' ...

The first argument to the "roundrectangle" draw operation will be two numbers showing the coordinates for the left and top of the rectangle. These two numbers should be separated by a comma only, no spaces.

The second argument is another comma separated pair of numbers showing the right and bottom coordinates for the rectangle.

The third argument is the horizontal and vertical radii of the rounded corner, again two numbers separated by a comma.

The only spaces there should be between the three arguments.

I'd suggest repairing those mistakes first, then try the command to see if there are other errors.

**Reply by matanshukry** [reply]
@GeeMack I don't think it matters, and I'm assuming this on the fact the command (and the spaces) came from the official documentation: 
https://imagemagick.org/script/command-line-processing.php

Either way it still doesn't work. Even the following:
```
magick -size 320x90 canvas:none -stroke snow4 -size 1x90 -tile gradient:white-snow4  -draw 'roundrectangle 16,5 304,85 20,40' -fill snow out.png
```

Results in the same error

**Reply by GeeMack** [reply]
Using IM 7.0.10-29 on Windows 10, if I put double quotes around that draw argument in that command instead of single quotes it produces the image below and stops producing the error.

    ... -draw "roundrectangle 16,5 304,85 20,40" ...

![out](https://user-images.githubusercontent.com/65475279/93356376-b5b02900-f804-11ea-858f-8e524673678a.png)

**Reply by matanshukry** [reply]
That solves it! double quotes it is!
Thanks!

**Comment by snibgo** [comment]
What version of IM, on what platform? Please always state this when asking questions.

If you use Windows, change the single quotes ' to double quotes ".
